src/app/core/services/basic-rundown-state.service.ts

```diff
@@ -129,4 +130,8 @@ export class BasicRundownStateService implements OnDestroy {
   public ngOnDestroy(): void {
     this.eventSubscriptions.forEach(eventSubscription => eventSubscription.unsubscribe())
   }
+
+  public getNonIdleRundown(): BasicRundown | undefined {
+    return this.basicRundowns.find(rundown => [RundownMode.ACTIVE, RundownMode.REHEARSAL].includes(rundown.mode))
+  }
```

Comment:
Instead of having a specific accessor on the state service, I would suggest to change to `getBasicRundowns()` and only have it return the basic rundowns array. Then where the `getNonIdleRundown` is used, can do the analysis that it needs.